src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -0,0 +1,702 @@
+  onCancel: null,
+  onApply: null,
+};
+
+const DateTimePicker = ({
+  defaultValue,
+  presets,
+  intervals,
+  relatives,
+  expanded,
+  disabled,
+  showRelativeOption,
+  onCancel,
+  onApply,
+  ...others
+}) => {
+  const [isExpanded, setIsExpanded] = useState(expanded);
+  const [customRangeKind, setCustomRangeKind] = useState(
+    showRelativeOption ? PICKER_KINDS.RELATIVE : PICKER_KINDS.ABSOLUTE
+  );
+
+  const [isCustomRange, setIsCustomRange] = useState(false);
+
+  const datePickerRef = React.createRef();
+
+  const [selectedPreset, setSelectedPreset] = useState(null);
+
+  const [currentValue, setCurrentValue] = useState(null);
+  const [humanValue, setHumanValue] = useState(null);
+
+  const [relativeValue, setRelativeValue] = useState(null);
+  const [absoluteValue, setAbsoluteValue] = useState(null);
+
+  const dateTimePickerBaseValue = {
+    kind: '',
+    preset: {
+      label: null,
+      offset: null,
+    },
+    relative: {
+      lastNumber: null,
+      lastInterval: null,
+      relativeToWhen: null,
+      relativeToTime: null,
+    },
+    absolute: {
+      startDate: null,
+      startTime: null,
+      endDate: null,
+      endTime: null,
+    },
+  };
+
+  useEffect(() => {
+    window.setTimeout(() => {
+      /* istanbul ignore next */
+      if (datePickerRef && datePickerRef.current) {
+        datePickerRef.current.cal.open();
+        // while waiting for https://github.com/carbon-design-system/carbon/issues/5713
+        // the only way to display the calendar inline is to reparent its DOM to our component
+        const wrapper = document.getElementById(`${iotPrefix}--date-time-picker__wrapper`);
+        if (typeof wrapper !== 'undefined' && wrapper !== null) {
+          const dp = document
+            .getElementById(`${iotPrefix}--date-time-picker__wrapper`)
+            .getElementsByClassName(`${iotPrefix}--date-time-picker__datepicker`)[0];
+          dp.appendChild(datePickerRef.current.cal.calendarContainer);
+        }
+      }
+    }, 0);
+  });
+
+  const parseValue = value => {
+    setCurrentValue(value);
+    let readableValue = '';
+    switch (value.kind) {
+      case PICKER_KINDS.RELATIVE: {
+        let endDate = moment();
+        if (value.relative.relativeToWhen !== '') {
+          endDate =
+            value.relative.relativeToWhen === RELATIVE_VALUES.YESTERDAY
+              ? moment().add(-1, INTERVAL_VALUES.DAYS)
+              : moment();
+          if (value.relative.relativeToTime) {
+            endDate.hours(value.relative.relativeToTime.split(':')[0]);
+            endDate.minutes(value.relative.relativeToTime.split(':')[1]);
+          }
+        }
+        const startDate = endDate
+          .clone()
+          .subtract(
+            value.relative.lastNumber,
+            value.relative.lastInterval ? value.relative.lastInterval : INTERVAL_VALUES.MINUTES
+          );
+        readableValue = `${moment(startDate).format('YYYY-MM-DD HH:mm')} to ${moment(
```

Comment:
translations.  either this needs to a formatted string `{0} to {1}`, or a function.
I would think that that the datetime mask would be a prop.  it won't always be that format.